Host side changes for Dispatch on Fabric

nhuang-tt opened the pull request:
### Ticket
https://github.com/tenstorrent/tt-metal/issues/18726

### Problem description
- Partial changes to enable fast dispatch on fabric
- This PR contains only Host side changes. Follow up PRs will be opened for more host + device changes

### What's changed
- Add topologies for 1CQ/2CQ N300/T3K and Galaxy
- Add `FabricMux` to the dispatch kernel factory. This creates and configures the FabricMux kernel. Optimization will need to be made in the future to select the optimal buffer size and number of buffers.
- Add functions to allocate the fabric mux in the `dispatch_core_manager`. The Mux is allocated based on the device, CQ, and tunnel index.
- Delete `FabricRouterVC`. This was for 2D fabric and is not used anymore
- Due to a limitation in the Fabric APIs, several changes in `topology.cpp` and `FDKernel` to enable querying the immediate neeighboring downstream/upstream device ID depending on the tunnel index.

### Checklist
APC
https://github.com/tenstorrent/tt-metal/actions/runs/15410273593
T3K
https://github.com/tenstorrent/tt-metal/actions/runs/15423560851
BH
https://github.com/tenstorrent/tt-metal/actions/runs/15424550546

nhuang-tt committed:
Add dispatch on fabric topologies

nhuang-tt committed:
Add fabric mux to dispatch core manager

nhuang-tt committed:
Move tunnel index to its own parameter

aagarwalTT commented on tt_metal/impl/dispatch/kernel_config/fabric_mux.cpp:
```diff
@@ -0,0 +1,185 @@
+    // Setup the buffer sizes depending on how many upstream/downstream kernels are connected
+    static_config_.num_full_size_channels = kernels_requiring_full_size_channel;
+    static_config_.num_header_only_channels = kernels_requiring_header_only_channel;
+    static_config_.buffer_size_bytes = sizeof(tt::tt_fabric::PacketHeader) + 4096;
```
can you try querying the packet header size from fabric context (`get_fabric_packet_header_size_bytes`)?

aagarwalTT commented on tt_metal/impl/dispatch/kernel_config/fabric_mux.cpp:
```diff
@@ -0,0 +1,185 @@
+    const auto& available_links = tt_fabric::get_forwarding_link_indices(device_id_, destination_device_id);
+    TT_ASSERT(!available_links.empty());
+    tt_fabric::append_fabric_connection_rt_args(
+        device_id_, destination_device_id, available_links[0], *program_, {logical_core_}, mux_rt_args_, GetCoreType());
```
for FD, can we always connect to the last available link idx? It will help to avoid the conflict with OPs

nhuang-tt replied:
Done

nhuang-tt committed:
Feedback
- Last last available link index
- Rename program factory to RelaxMux
- Query packet header size with Fabric API

nhuang-tt commented on tt_metal/impl/dispatch/kernel_config/fabric_mux.hpp:
```diff
@@ -0,0 +1,115 @@
+// Servicing Device Id: The downstream tunnel index
+// d2h: True means this MUX is for returning data from device to host. Servicing device Id is ignored if this is true.
+//
+class FabricMux : public FDKernel {
```
Renamed to `RelayMux`
